dnf/cli/commands/__init__.py

```diff
@@ -900,11 +909,14 @@ def _hcmd_userinstalled(self):
         pkgs = tuple(self.base.iter_userinstalled())
         return self.output.listPkgs(pkgs, 'Packages installed by user', 'nevra')
 
-    def _convert_tids(self, merged=set()):
+    def _convert_tids(self, merged=set(), require_one=False, require_one_msg=''):
         """Convert commandline arguments to transaction ids"""
+
         def str2tid(s):
-            if s.startswith('last'):
-                s = s[4:] if s != 'last' else '0'
+            if s == 'last':
+                s = '0'
+            elif s.startswith('last-'):
+                s = s[4:]
             tid = int(s)
```

Comment:
Throw of exception may happen and it is desired behaviour. Exception is handled by caller. Handling of exception depends on caller.